lib/ts_utils/requirements.py

```diff
@@ -13,14 +12,14 @@
 
 def get_external_stub_requirements(distributions: Iterable[str] = ()) -> set[Requirement]:
     if not distributions:
-        distributions = os.listdir(STUBS_PATH)
+        distributions = [distribution.name for distribution in STUBS_PATH.iterdir()]
 
     return set(itertools.chain.from_iterable([read_dependencies(distribution).external_pkgs for distribution in distributions]))
 
 
 def get_stubtest_system_requirements(distributions: Iterable[str] = (), platform: str = sys.platform) -> list[str]:
     if not distributions:
-        distributions = os.listdir(STUBS_PATH)
+        distributions = [distribution.name for distribution in STUBS_PATH.iterdir()]
```

Comment:
This seems like a straight up downgrade.
[os-listdir (PTH208)](https://docs.astral.sh/ruff/rules/os-listdir/#os-listdir-pth208)

Only one other change of `os.listdir` --> `Path.iterdir` seems like an actual improvement. (in `tests/mypy_test.py`)